server/session.go

```
@@ -0,0 +1,217 @@
+	if sessionTools == nil {
+		sessionTools = make(map[string]ServerTool)
+	}
+
+	for _, tool := range tools {
+		sessionTools[tool.Tool.Name] = tool
+	}
+
+	session.SetSessionTools(sessionTools)
+
+	// Send notification only to this session
+	s.SendNotificationToSpecificClient(sessionID, "notifications/tools/list_changed", nil)
```

golangci-lint (1.64.8)

Error return value of `s.SendNotificationToSpecificClient` is not checked

(errcheck)